src/openforms/js/components/admin/form_design/variables/VariablesTable.js

```diff
@@ -0,0 +1,129 @@
+        defaultMessage: 'Are you sure you want to delete this variable?',
+    });
+
+    const {availablePrefillPlugins} = useContext(PluginsContext);
+    const prefillPluginChoices = availablePrefillPlugins.map((plugin) => [plugin.id, plugin.label]);
+    const [prefillAttributeChoices, setPrefillAttributeChoices] = useState([]);
+
+    const onValueChanged = (e) => {
+        onChange(variable.key, e.target.name, e.target.value);
+    };
+
+    const updateKey = (e) => {
+        // Key creation taken from FormIO
+        let updatedKey = _.camelCase(variable.name).replace(/^[0-9]*/, '');
+        onChange(variable.key, 'key', updatedKey);
+    }
+
+    const {loading} = useAsync(async () => {
+        setPrefillAttributeChoices([]);
+        // Load the possible prefill attributes
+        if (!variable.prefillPlugin) return;
+
+        const url = `/api/v1/prefill/plugins/${variable.prefillPlugin}/attributes`;
+        const response = await get(url);
+        if (!response.ok) {
+            console.error(response.data);
+        }
+
+        setPrefillAttributeChoices(response.data.map((attribute) => [attribute.id, attribute.label]))
+
+    }, [variable.prefillPlugin])
+
+    return (
+        <tr>
+            <td><DeleteIcon onConfirm={() => onDelete(variable.key)} message={deleteConfirmMessage} /></td>
+            <td><TextInput name="name" value={variable.name} onChange={onValueChanged} onBlur={updateKey} noVTextField={true} /></td>
+            <td><TextInput name="key" value={variable.key}  noVTextField={true} disabled={true} /></td>
+            <td><Select name="prefillPlugin" choices={prefillPluginChoices} value={variable.prefillPlugin} onChange={onValueChanged} allowBlank /></td>
+            <td><Select name="prefillAttribute" choices={prefillAttributeChoices} value={variable.prefillAttribute} onChange={onValueChanged} disabled={loading || (!variable.prefillPlugin)}/></td>
+            <td><Select name="dataType" choices={DATATYPES_CHOICES} translateChoices value={variable.dataType} onChange={onValueChanged} /></td>
+            <td><Checkbox name="" label="" checked={variable.isSensitiveData} onChange={(e) => onValueChanged({target: {name: 'isSensitiveData', value: !variable.isSensitiveData}})} /></td>
+            <td><TextInput name="initialValue" value={variable.initialValue || ''} onChange={onValueChanged} noVTextField={true} /></td>
+            <td><TextInput name="dataFormat" value={variable.dataFormat} onChange={onValueChanged} noVTextField={true} /></td>
+        </tr>
+    );
+};
+
+const VariablesTable = ({variables, editable, onChange, onDelete}) => {
+    return (
+        <table className="variables-table">
+            <thead>
+                <tr>
+                    { editable && <th className="variables-table__header"/> }
+                    <th className="variables-table__header"><FormattedMessage defaultMessage="Name" description="Variable table name title" /></th>
```

Comment:
I wonder if we can re-use django's classnames instead, especially considering we have https://github.com/open-formulieren/open-forms/tree/master/src/openforms/js/components/admin/tables

Reply:
Ah nice! I didn't notice these components. I will try to update the tables to use them